src/track/track.cpp

```diff
@@ -368,6 +368,20 @@ void Track::setDateAdded(const QDateTime& dateAdded) {
     return m_record.setDateAdded(dateAdded);
 }
 
+QDateTime Track::getLastPlayedDate() const {
+    QMutexLocker lock(&m_qMutex);
+    return m_lastPlayed;
+}
+
+void Track::setLastPlayedDate(const QDateTime& playedDate) {
+    QMutexLocker lock(&m_qMutex);
+    if (playedDate == m_lastPlayed) {
```

Comment:
There is a small utility function for setting track properties: `compareAndSet()`

Reply:
done